### PR TITLE
Increase documentation coverage

### PR DESCRIPTION
**Pull request by komish**
The goal of this PR is just to increase documentation coverage. 

This PR does add a net-new doc on how to implement a check, with a clear note indicating that we may reject a PR for a check given our goal of ensuring this tool accurately reflects the pipeline's validation.

Finally, this adds several comments throughout, including package comments which will show up in pkg.go.dev.

Signed-off-by: Jose R. Gonzalez <josegonzalez89@gmail.com>

### CODE REVIEW

**Review by samira-barouti — APPROVED** [review]

**Comment by komish** [comment]
An extra note, this also removes the `CheckInfo` struct which was previously used in user responses, but we've since standardized on that response and so this CheckInfo struct is no longer in use.

File: certification/internal/shell/validate_operator_bundle.go [hunk 1]
```diff
@@ -6,13 +6,15 @@ import (
...
+// as executed by `operator-sdk bundle validate`
 type ValidateOperatorBundleCheck struct {
 }
 
 func (p ValidateOperatorBundleCheck) Validate(bundle string) (bool, error) {
 	report, err := p.getDataToValidate(bundle)
 	if err != nil {
-		log.Error("Error will executing operator-sdk validate bundle: ", err)
+		log.Error("Error will executing operator-sdk bundle validate: ", err)
 		return false, err
```
Comment (tonytcampbell) [inline certification/internal/shell/validate_operator_bundle.go:18]: Typo:  "Error while executing....."

Comment (komish) [reply certification/internal/shell/validate_operator_bundle.go:18]: Good catch, fixed!

File: docs/dev/README.md [hunk 2]
```diff
@@ -5,91 +5,27 @@ operator projects to see if they pass Red Hat operator certification
 requirements.
 
 The project will include a commandline interface that will accept your operator
-bundle image as an input, and will run validate that your operator bundle
-complies with a configurable set of checks.
+bundle or container image as an input, and will run validate that your container
+image or operator bundle complies with a series of validations.
```
Comment (tonytcampbell) [inline docs/dev/README.md:9]: is validation a synonym of check at the end of Line 9?  I think we should be consistent with out terms and point out if we are talking about something different than checks.

**Review by tonytcampbell — COMMENTED** [review]
Few nits, but looks good outside of that.

**Review by bcrochet — APPROVED** [review]
/lgtm

**Pull request merged**